appcelerator/titanium_mobile
# [6_1_X][TIMOB-24712] Android: Fixed TabGroup to not override ActionBar's "colorPrimary" XML theme.

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24712

**Notes:**

- Issue was caused by line number 342 in old code. It was returning a default color when no color was assigned to the Tab or TabGroup. Should not change the background drawable at all if no color was assigned in order to respect the default theme.
- Removed the "actionBar.setBackgroundDrawable()" line. It wasn't working to begin with (tested in 6.0.2) because you can't share the same drawable instance between views. Plus, the developer probably wouldn't want the TabGroup color to override the ActionBar color set in the XML theme. I talked to Lokesh about this and he agrees.

**Test Case:**

1. Add the below files to a classic Titanium app.
2. Build and run on an Android device.
3. Verify that the top ActionBar (ie: title bar) is **blue**. (Comes from the XML theme file.)
4. Verify that the tabs are **red**. (This is assigned in JavaScript.)
5. Tap the button that is centered inside of one of the tabs.
6. A new window is displayed.
7. Press the back button to exit the 2nd window.
8. Verify that you were returned to the 1st window without issue. (This tests for [TIMOB-24367](https://jira.appcelerator.org/browse/TIMOB-24367).)

**app.js:**
```javascript
function createTab(title, buttonText) {
	var win = Ti.UI.createWindow({
		title: title,
		backgroundColor: '#fff'
	});

	var button = Ti.UI.createButton();
	button.title = buttonText;
	button.addEventListener("click", function(e) {
		var newWindow = Ti.UI.createWindow({
			backgroundColor: "blue",
			title: buttonText + " Window",
		});
		newWindow.add(Ti.UI.createLabel({text: 'Tap Back Button'}));
		newWindow.open();
	});
	win.add(button);

	var tab = Ti.UI.createTab({
		title: title,
		backgroundColor: "red",
		backgroundFocusedColor: "red",
		window: win,
	});
	return tab;
}

var tabGroup = Ti.UI.createTabGroup();
tabGroup.addTab(createTab("Tab 1", "Button 1"));
tabGroup.addTab(createTab("Tab 2", "Button 2"));
tabGroup.open();
```

**./platform/android/res/values/builtin_themes.xml:**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<resources>
	<style name="tracker" parent="@style/Theme.AppCompat">
		<item name="colorPrimary">#0000FF</item>
		<item name="colorAccent">#FFFFFF</item>
		<item name="android:statusBarColor">#004486</item>
		<item name="android:navigationBarColor">#004486</item>
	</style>
</resources>
```
